ecs_deploy/cli.py

```diff
@@ -173,13 +178,18 @@ def run(cluster, task, count, command, env, secret, region, access_key_id, secre
         action = RunAction(client, cluster)
 
         td = action.get_task_definition(task)
+        td.set_images(tag, **{key: value for (key, value) in image})
         td.set_commands(**{key: value for (key, value) in command})
-        td.set_environment(env)
-        td.set_secrets(secret)
+        td.set_environment(env, exclusive_env)
+        td.set_secrets(secret, exclusive_secrets)
+        td.set_role_arn(role)
 
         if diff:
             print_diff(td, 'Using task definition: %s' % task)
 
+        if image or tag or role:
+            td = create_task_definition(action, td)
```

Comment:
If you create a new task definition here, I think it should be deregistered, after the task had been successfully executed. Otherwise you end up with `n` active task definitions in ECS. What do you think? As in the `deploy` action, this could be controllable via a cli parameter

Reply:
Added but I highly discourage that feature. You can end using very old task definitions or end without any active one. Very risky